crates/cow-amm/src/cow_amm_constant_product_factory.rs

```diff
@@ -0,0 +1,30 @@
+use {
+    crate::{cow_amm::CowAmm, indexer::CowAmmRegistry},
+    shared::impl_event_retrieving,
+};
+
+impl_event_retrieving! {
+    pub Contract for contracts::cow_amm_constant_product_factory
+}
+
+pub struct CowAmmConstantProductFactoryHandler;
```

Comment:
This seems like it would have to be a trait to cover work with the different implementations.